dFusion/dFusionSpec.md

```diff
@@ -50,43 +48,48 @@ function appendOrders( bytes32 [] orders){
+This function will simply update an orderHashSha variable, which is chaining all orders with a valid signature. This function is callable by any party. However, it is expected that “decentralized operators” accept orders from users, bundle them and then include them all together into the function. 
+
+Notice, that the orders are only sent over as transaction payload, but will not be “stored” in the EVM.
 
-### Transition function from sha to Pederson hashes 
+Also notice, that we allow orders, which might not be covered by any balance of the order sender. These orders will be sorted out later in the settlement of an auction.
 
-In the first step, the orders are hashed together using sha. This makes sense as sha is very cheap on the evm. However, sha is very “expensive” in snarks and hence we are forced to recalculate the hashes in Pederson hashes. 
+### Transition function from sha to Pedersen hashes 
+
+In the first step, the orders are hashed together using SHA256 since this is very cheap on the EVM. However, SHA256 is very “expensive” in snarks. We therefore translate the resulting orderHash into a pedersen hash after order collection for a batch has finished. 
 
 We will use a snark to do this job:
 ```
 Snark - TransitionHashes&Validation ( public input: orderHashSha,
 					Private input: [orders])
-					Output: orderHashPederson
+					Output: orderHashPedersen
 ```
 The transitionHashes&Validation snark will do the following checks:
-- Verify the private input by recalculating the sha of all orders and comparing it to the public input `orderHashSha`.
-- Iterate over all orders again and hash them sequencially using the Pederson hash. Use this hash as output.
-
-Notice that we allow orders, which might not be covered by any balance of the order sender. These orders will be sorted out later in the settlement of an auction.
+- Verify the private input by recalculating SHA256 of all orders and comparing it to the public input `orderHashSha`.
+- Iterate over all orders again and hash them sequencially using the Pedersen hash. Use this hash as output.
 
-In the anchor contract, we have the following functionality for this process:
+Since computing the actual snark proof is very time-intense we optimistically accept state transitions that provide a significant bond instead of the actual proof.
 
-Anyone can propose a transition to the anchor contract by providing the required information and by providing a very significant bond. It is not required to provide the snark in the first place:
+Anyone can propose a transition to the anchor contract by providing the required information:
 
 ```js
-Function submitTransitionInformation( bytes32 oldstate, bytes32 newstate)
+Function submitTransitionInformation(bytes32 orderHashPedersen)
```

Comment:
I prefer to reference to a unique old state, as transactions might be in flight for longer than one auction period.

Reply:
The let's reference a batchId instead. In the current model it suggests that the solution provider can provide the old state, however the old state is chosen/fixed by the smart contract.

Reply:
Please do not use the batchID, because we want to support the branching mechanism

If you want you can use the batchID and I will update it later @fleupold

Reply:
used branchId instead